File: plugins/backend/app/controllers/backend/IndexAction.java
package controllers.backend;

import com.avaje.ebean.Ebean;
import models.AppFuncTree;
import org.springframework.stereotype.Service;
import play.Logger;
import play.Play;
import play.mvc.Controller;
import play.mvc.Result;

import java.util.List;
import java.util.concurrent.atomic.AtomicInteger;

import static plugins.freemarker.Freemarker._;
import static plugins.freemarker.Freemarker.view;

/**
 * Created by zhangmeng on 16-6-27.
 */
@Service
public class IndexAction extends Controller {

    String appid = Play.application().configuration().getString("appid");

    public AtomicInteger ai = new AtomicInteger(0);

    public Result login() {
        Logger.info("login index..");
        return ok(view("backend/login.ftl"));
    }

    public Result index(){
        List<AppFuncTree> appFuncTrees = Ebean.find(AppFuncTree.class).where().eq("appId", appid).findList();
        return ok(view("backend/index.ftl", _("title", "PLATFORM"), _("trees", appFuncTrees)));
    }


}

File: plugins/backend/app/controllers/backend/IndexAction.scala
package controllers.backend

import commons.Eithers
import play.api.Play
import play.api.mvc.{Action, Controller}
import play.core.j.JavaParsers
import play.libs.Scala
import plugins.freemarker.Freemarker._
import plugins.spring.Spring
import services.backend.ILogin

/**
 * Created by zhangmeng on 16-9-7.
 */
object IndexAction extends Controller{
  val loginSer = Play.current.configuration.getString("loginService")

  val loginService = Spring.getBean(loginSer.getOrElse("")).asInstanceOf[ILogin]

  def login = Action {
    val html = loginService.loginHtml()
    Ok(view(html))
  }

  def loginInvoke = Action(parse.json){ request =>
    val json = request.body
    val username = (json \ "username").as[String]
    val map = Map("username" -> username)
    val resp = loginService.loginInvoke(Scala.asJava(map))
    Ok()
  }

  def index = Action{
    Ok
  }
}
